java/org/apache/tomcat/util/digester/EnvironmentPropertySource.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.tomcat.util.digester;
+
+import java.security.Permission;
+
+import org.apache.tomcat.util.IntrospectionUtils;
+import org.apache.tomcat.util.security.PermissionCheck;
+
+public class EnvironmentPropertySource implements IntrospectionUtils.SecurePropertySource {
```

Comment:
Since you are doing a bit of re-factoring, could you add some javadoc? Your use of the class suggests that you understand its use and the need for its existence. No need for anything fancy. Just a one or two sentence description and maybe a reference to the users manual if appropriate.

Reply:
Super idea will do it. Give me time til Friday.

Reply:
Can you take a look please